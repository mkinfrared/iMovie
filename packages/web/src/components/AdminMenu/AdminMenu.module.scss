@import "src/styles/colors";
@import "src/styles/animations";

.AdminMenu {

  .list {
    width: 250px;
  }
}
