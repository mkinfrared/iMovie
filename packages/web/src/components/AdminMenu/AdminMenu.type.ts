export interface AdminMenuProps {
  isOpen: boolean;
  onClose: () => void;
}

export interface Navigation {
  text: string;
  icon: JSX.Element;
  path: string;
}
