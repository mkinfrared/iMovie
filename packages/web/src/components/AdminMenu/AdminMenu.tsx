import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import MailIcon from "@material-ui/icons/Mail";
import InboxIcon from "@material-ui/icons/MoveToInbox";
import React, { useMemo } from "react";

import css from "components/Header/Header.module.scss";

import { AdminMenuProps, Navigation } from "./AdminMenu.type";

const navigation: Navigation[] = [
  {
    text: "Lorem",
    icon: <InboxIcon />
  },
  {
    text: "Ipsum",
    icon: <MailIcon />
  }
];

const AdminMenu = ({ isOpen, onClose }: AdminMenuProps) => {
  const list = useMemo(
    () =>
      navigation.map(({ icon, text }) => (
        <ListItem button key={text}>
          <ListItemIcon>{icon}</ListItemIcon>
          <ListItemText primary={text} />
        </ListItem>
      )),
    []
  );

  return (
    <Drawer open={isOpen} onClose={onClose}>
      <List className={css.list}>{list}</List>
    </Drawer>
  );
};

export default React.memo(AdminMenu);
