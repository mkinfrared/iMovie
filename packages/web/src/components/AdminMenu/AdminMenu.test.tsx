import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import AdminMenu from "./AdminMenu";

jest.mock("components/AdminMenuLink", () => (props: any) => (
  <div data-testid="components/AdminMenuLink">
    <code>{JSON.stringify(props)}</code>
  </div>
));

describe("<AdminMenu />", () => {
  let Component: ReactElement;

  const onClose = jest.fn();

  beforeEach(() => {
    Component = <AdminMenu isOpen onClose={onClose} />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
