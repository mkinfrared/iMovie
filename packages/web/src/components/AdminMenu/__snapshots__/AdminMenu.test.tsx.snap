// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminMenu /> should match a snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
